source/grid/grid_tools.cc

```diff
@@ -4128,6 +4128,76 @@ namespace GridTools
       }
   }
 
+  namespace internal
+  {
+    // Split get_subdomain_association() for p::d::T since we want to compile it
+    // in 1D but none of the p4est stuff is available in 1D.
+    template <int dim, int spacedim>
+    void
+    get_subdomain_association(
```

Comment:
Move this function further into an anonymous namespace. The function does not need to be exported from this .cc file.

Reply:
Fixed!